### PR TITLE
Feat: pagination and results selector

### LINKED ISSUE #57: Design Pagination and Results Selector
## 📘 **Description**

A pagination layout is essential to enhance user experience in the Marketplace by dividing product listings into manageable sections. This visual structure prevents overwhelming users with a continuous scroll of items, instead allowing them to navigate through pages. Additionally, a selector to customize the number of items displayed per page improves flexibility. Although the pagination won't be functional yet, designing it now aligns the interface with future usability goals and ensures consistency in the Marketplace layout.

## 🔢 **Steps**

1. Create a new file named `products-pagination.tsx` inside the `components/marketplace` directory. If the directory does not exist, create it.
2. Name the component ProductsPagination.
3. Implement the pagination and selector layout using **shadcn/ui** components:
    - [Pagination](https://ui.shadcn.com/docs/components/pagination)
    - [Select](https://ui.shadcn.com/docs/components/select)
4. Ensure the layout is responsive and integrates seamlessly with the Marketplace design.

## ✅ **Acceptance Criteria**

- Include a pagination component.
- Include a results selector with options to display 10, 20, 30, and 50 items per page.
- The design should be responsive.

## 🌎 **References**

- Reference attached images for design inspiration, but the final implementation does not need to be identical.

image (11)

image (12)

## 📜 **Additional Notes**

- Focus solely on the visual implementation; functionality is not required at this stage.

⚠ **Do not apply until the ODHack begins.**
⚠ **Read our guidelines before applying.**

### PR DESCRIPTION
**Pull request by armandocodecr**
# 📝 PAGINATION AND RESULT SELECTOR

## 🛠️ Issue
- Closes #57 

## 📖 Description
- Implemented the UI for the future marketplace product page.

## ✅ Changes made
1. I installed the requested shadcn components
2. Create the ProductsPagination component which in turn renders the pagination component and the results selector component.
3. I used the above mentioned component inside the marketplace page. Specifically below the list of products.

## 🖼️ Media (screenshots/videos)
- [SAFESWAP - ISSUE #57 | DEMO VIDEO](https://youtu.be/8gS9EgDAwsw)

## 📜 Additional Notes
- None

### CODE REVIEW

File: frontend/app/components/marketplace/productos-pagination.tsx
Comment (derianrddev) [inline frontend/app/components/marketplace/productos-pagination.tsx:1]: Rename to products-pagination

File: frontend/app/components/marketplace/index.ts
Comment (derianrddev) [inline frontend/app/components/marketplace/index.ts]: Can you add a line at the end of the file, please?

File: frontend/app/components/marketplace/productos-pagination.tsx
Comment (derianrddev) [inline frontend/app/components/marketplace/productos-pagination.tsx]: Change the width to w-[70px] in the `SelectTrigger`

File: frontend/app/components/marketplace/productos-pagination.tsx
Comment (derianrddev) [inline frontend/app/components/marketplace/productos-pagination.tsx]: Add min-w-[70px] here

File: frontend/app/components/marketplace/productos-pagination.tsx
Comment (derianrddev) [inline frontend/app/components/marketplace/productos-pagination.tsx]: It seems that `SelectLabel` is not being used. Please remove the import.

**Review by derianrddev — CHANGES_REQUESTED** [review]

**Comment by derianrddev** [comment]
@armandocodecr Good job! Just some minor comments.

File: frontend/app/components/marketplace/productos-pagination.tsx
Comment (armandocodecr) [reply frontend/app/components/marketplace/productos-pagination.tsx]: Do you want me to add it to the “SelectTrigger” component?

Comment (derianrddev) [reply frontend/app/components/marketplace/productos-pagination.tsx]: In the `SelectContent`

**Comment by armandocodecr** [comment]
@derianrddev I have made the requested changes. What do you think?

**Review by derianrddev — APPROVED** [review]

**Comment by derianrddev** [comment]
@armandocodecr It looks good to me, we just need @danielcdz's approval to merge it.

**Review by danielcdz — APPROVED** [review]
LGTM

**Comment by danielcdz** [comment]
@armandocodecr can you resolve the conflicts? after that we can merge, nice job, thank you!

**Comment by armandocodecr** [comment]
@danielcdz I have just resolved the conflicts. I remain attentive in case you need me to make any more changes. :)

**Comment by armandocodecr** [comment]
@danielcdz I have just resolved the conflicts. I remain attentive in case you need me to make any more changes. :)

**Pull request merged**
